pallets/invulnerables/src/lib.rs

```diff
@@ -56,12 +60,26 @@ pub mod pallet {
         /// Overarching event type.
         type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 
+        /// The currency mechanism.
+        type Currency: ReservableCurrency<Self::AccountId>;
```

Comment:
We should not need the Currency type I think

Reply:
done